File: Frontend/src/app/services/movie.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Movie } from '../model/movie';

@Injectable({
  providedIn: 'root'
})
export class MovieService {

  baseUrl = 'http://localhost:9090/api/v1/moviebooking';

  constructor(private httpClient: HttpClient) { }

  //getAllMovies() api
  public getAllMovies() {
    return this.httpClient.get(this.baseUrl + '/all');
  }

  //addMovie() api
  public addMovie(movie: Movie) {
    return this.httpClient.post<Movie>(this.baseUrl + '/addmovie', movie);
  }

  //getMovieById()
  // public getMovieById(movieId: number) {
  //   return this.httpClient.get(this.baseUrl + '/movie/' + movieId);
  // }

  //deleteMovie
  public deleteMovie(movieId: number) {
    return this.httpClient.delete(this.baseUrl + '/delete/' + movieId);
  }

}

File: Backend/movie-booking-app/src/main/java/com/booking/service/MovieService.java
package com.booking.service;

import java.util.List;

import com.booking.exception.MovieIdAlreadyExistsExceptions;
import com.booking.model.Movie;

public interface MovieService {

	List<Movie> getAllMovies();

	Movie getMovieById(int movieId);

	void deleteMovieById(int movieId);

	Movie addMovie(Movie movie) throws MovieIdAlreadyExistsExceptions;

}
